lib/s3routes/routesUtils.ts

```diff
@@ -846,7 +846,7 @@ export function redirectRequest(
     }
     let redirectLocation = justPath ? `/${redirectKey}` :
         `${redirectProtocol}://${redirectHostName}/${redirectKey}`;
-    if (!redirectKey && redirectLocationHeader) {
+    if (!redirectKey && redirectLocationHeader && redirectLocation.length > 1) {
```

Comment:
I think it would be more easily understood this way:

```suggestion
    if (!redirectKey && redirectLocationHeader && redirectLocation !== '/') {
```